packages/expect/src/types.ts

```diff
@@ -86,6 +86,7 @@ export type ExpectedAssertionsErrors = Array<{
 
 export interface BaseExpect {
   assertions(numberOfAssertions: number): void;
+  customEqualityTesters: Array<Tester>;
```

Comment:
What about `addEqualityTesters(testers: Array<Tester>): void;`? Just an idea for discussion.

Also note that a type tests will be needed. No rush with that, just add it to the TODO list. These should be similar to `expect.extend()`tests which live in [packages/expect/\_\_typetests\_\_/expect.test.ts](https://github.com/facebook/jest/blob/main/packages/expect/__typetests__/expect.test.ts)

Reply:
Added type tests to the TODO!

Yea, that might be a better API. The current proposed API is a bit unique in the BaseExpect API and so kind of sticks out. The current proposal does allow overwriting the custom equality testers so you can remove one if you'd like but perhaps that is unnecessary and test suites that require that should perhaps belong in separate files anyway.

If others agree with `addEqualityTesters`, let me know and I'll make the change!

Reply:
The customizability ability is nice but probably leaves a lot of room for footguns, like the async test example you mentioned in the other comment.

I'm not aware of any existing expect API that scopes things to globa/describe/test blocks. Would that be new functionality?

I wonder if we could start with just a simple `addEqualityTesters` to the global scope, and people can use Jest's existing test-file boundaries to encapsulate where they add custom testers (either by calling a helper method in each test file, or using a custom jest setup script, or writing a custom environment).

Reply:
Updated API to `addEqualityTesters`